Trait `std::fmt::Debug` is not implemented for `sigset_t` in libc

Before posting the bug, take a look at the
[FAQ](https://github.com/eclipse-iceoryx/iceoryx2/blob/main/FAQ.md)
for a possible solution.

## Required information

**Operating system:**
Linux, Kernel 6.16.10

**Rust version:**
rustc 1.90.0

**Cargo version:**
cargo 1.90.0

**iceoryx2 version:**
`main` branch

**Detailed log output:**
```
error[E0277]: `libc::unix::linux_like::linux::gnu::b64::sigset_t` doesn't implement `std::fmt::Debug`
  --> iceoryx2-bb/posix/src/signal_set.rs:22:5
   |
20 | #[derive(Debug)]
   |          ----- in this derive macro expansion
21 | pub struct SignalSet {
22 |     signal_set: posix::sigset_t,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::fmt::Debug` is not implemented for `libc::unix::linux_like::linux::gnu::b64::sigset_t`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `iceoryx2-bb-posix` (lib) due to 1 previous error
```

**Observed result or behavior:**
Compilation error

**Expected result or behavior:**
Compiles without error

**Conditions where it occurred / Performed steps:**
Compile command: `cargo build --features "libc_platform" -p iceoryx2-bb-posix`

Solution is to enable the feature `extra_traits` for the libc dependency. See https://github.com/rust-lang/libc/blob/4d939b0d26d97f8794f14bdf2638e39bf19d4af1/README.md?plain=1#L49

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -105,7 +105,7 @@ colored = { version = "2.1" }
 generic-tests = { version = "0.1.2" }
 human-panic = { version = "2.0.2" }
 lazy_static = { version = "1.4.0" }
-libc = { version = "0.2.169" }
+libc = { version = "0.2.169", features = ['extra_traits'] }
 log = { version = "0.4.21" }
 once_cell = { version = "1.19.0" }
 postcard = { version = "1.1.3", default-features = false, features = ["alloc"] }

--- a/doc/release-notes/iceoryx2-unreleased.md
+++ b/doc/release-notes/iceoryx2-unreleased.md
@@ -35,6 +35,9 @@
     [#1045](https://github.com/eclipse-iceoryx/iceoryx2/issues/1045)
 * LocalService in C language binding uses IPC configuration
     [#1059](https://github.com/eclipse-iceoryx/iceoryx2/issues/1059)
+* Trait `std::fmt::Debug` is not implemented for `sigset_t` in libc
+    [#1087](https://github.com/eclipse-iceoryx/iceoryx2/issues/1087)
+*
 
 ### Refactoring
 

